extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use baseInstance file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Ad display state.
+ * @enum {number}
+ */
+export const AdDisplayState = {
```

Comment:
right. ResourceState can't cover every state I need. Basicly I need a `loaded-with-content` and a `no-content`. ResourceState put these two under the `LAYOUT_COMPLETE` category. We have to introduce extra state here, and I would prefer introducing another state instead of introducing another boolean.